src/commons/build-partial-update-state-command.ts

```diff
@@ -11,12 +11,14 @@ import { HistoryStringBuilder } from './history-string-builder';
  * @param currentTimestamp - timestamp of now in ms
  * @param interventionName - optional intervention name if the event was a fraud intervention
  * @param interventionEvent - intervention type
+ * @param previousAppliedAt - timestamp of previous intervention
  */
 export const buildPartialUpdateAccountStateCommand = (
   finalState: StateDetails,
   eventName: EventsEnum,
   currentTimestamp: number,
   interventionEvent: TxMAIngressEvent,
+  previousAppliedAt: number,
```

Comment:
hey @amd-gds - I think Sonar Cloud is reporting code smell on on this function for cognitive complexity. https://sonarcloud.io/project/issues?resolved=false&sinceLeakPeriod=true&types=CODE_SMELL&pullRequest=169&id=govuk-one-login_account-interventions-service&open=AY0b4fJ6EG2C8jSkPVd0 it would good to see if there is any way we can break it down a bit, or take the logging of the metric out of here, as it is now quite hard to read

Reply:
hey Chiara - just noticed... will have a look

Reply:
thanks 🙌